### DISCUSSION TITLE
'Alarm control panel' dialog don't honor supported features

### DISCUSSION THREAD

**Discussion by javicalle** [home-assistant/frontend#11287] [Other feature requests] [answered]
## The request

The 'more-info' dialog for the 'alarm control panel', always show 2 bottons (`alarm_arm_home` and `alarm_arm_away`).
It will be great if the dialog can honor the device `supported_features` property and show only relevant buttons.

This case is different from the lovelace `alarm-panel` card, which allows you to configure the buttons that you want to show on the card (and they do not have to be all the actions available).

## The alternatives

I'm not sure which way to go in this case.
My first impulse would be to follow the example of the [`MoreInfoCover`](https://github.com/home-assistant/frontend/blob/dev/src/dialogs/more-info/controls/more-info-cover.js) and the [`CoverEntity`](https://github.com/home-assistant/frontend/blob/dev/src/util/cover-model.js), but it doesn't seem like a very common model.
I think the model is more in the line of using the [`supportsFeature`](https://github.com/home-assistant/frontend/blob/dev/src/common/entity/supports-feature.ts) function, but I get lost trying to follow the implementation. I think a good example candidate to follow may be [`MoreInfoVacuum`](https://github.com/iantrich/home-assistant-polymer/blob/4ad0ad75c5d0894b778b7999c9367b4754e29acc/src/dialogs/more-info/controls/more-info-vacuum.js)

## Additional information

Alarm control panel dialog:
* https://github.com/home-assistant/frontend/blob/1b970e5a66c05fcd36df392eb0d6c2f6f76d3d5d/src/dialogs/more-info/controls/more-info-alarm_control_panel.js#L138-L168

Alarm control Panel supported features attribute:
* https://github.com/home-assistant/core/blob/d587f134ca7d57bf2948c711fc9bd330aa48aa0c/homeassistant/components/alarm_control_panel/__init__.py#L170-L173

**Comment by Mincka** [comment]
Great suggestion. My alarm was wrongfully triggered recently because it was not activated in the proper mode from this dialog. I really wished that other available modes would be also shown in this dialog. In my case, I missed the `arm_custom_bypass` but all of them makes sense, including `trigger` that would allow us to get a "Panic" button in the dialog.

```
class AlarmControlPanelEntityFeature(IntEnum):
    """Supported features of the alarm control panel entity."""

    ARM_HOME = 1
    ARM_AWAY = 2
    ARM_NIGHT = 4
    TRIGGER = 8
    ARM_CUSTOM_BYPASS = 16
    ARM_VACATION = 32
```

**Comment by melyux** [comment]
Yeah I don't know why @zsarnett converted this to a discussion, it's a bug. The card shows the supported buttons while the more info dialog doesn't. There's a mismatch.

**Comment by karwosts** [comment, marked as answer]
See https://github.com/home-assistant/frontend/pull/15452

**Reply by Mincka** [reply]
Thank you!
